### PR TITLE
Descheduler deployment and running e2e test automation

### PR DESCRIPTION
**Pull request by AdityaHonkalas**
Signed-off-by: Aditya Honkalas <aditya.honkalas@ibm.com>

### CODE REVIEW

File: playbooks/roles/ocp-descheduler/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+- fail:
+    msg: "Please set descheduler_e2e_repo variable with the e2e repo URL."
+  when: descheduler_e2e_repo == "" or descheduler_e2e_repo == None
+
+- fail:
+    msg: "Please set the environment variables GITHUB_USERNAME and GITHUB_ACCESS_TOKEN on your host."
+  when: descheduler_github_username == "" or descheduler_github_access_token == ""
+
+# Cluster health check block
+- name: Check the cluster health
```
Comment (pravin-dsilva) [inline playbooks/roles/ocp-descheduler/tasks/main.yaml:19]: there is an existing role [check-cluster-health](https://github.com/ocp-power-automation/ocp4-playbooks-extras/tree/main/playbooks/roles/check-cluster-health). Can you re-use that?

File: playbooks/roles/ocp-descheduler/tasks/main.yaml [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        path: "{{ descheduler_e2e_dir }}/test/extended/workloads/kube_descheduler_operator.go"
+        regexp: 'imageInfo\:\s*\"[a-z0-9\.\/\:\-]*\"'
+        replace: "imageInfo:  \"{{ descheduler_catalogsource_image }}\""
+
+    - name: Compile the go modules using make command
+      make:
+        chdir: "{{ descheduler_e2e_dir }}"
+        target: build
+      
+    - name: Create a bin executable extended-platorm-tests
```
Comment (pravin-dsilva) [inline playbooks/roles/ocp-descheduler/tasks/main.yaml:163]: looks like this is not required.

**Review by pravin-dsilva — CHANGES_REQUESTED** [review]

File: playbooks/roles/ocp-descheduler/tasks/main.yaml (hunk 1 above)
Comment (AdityaHonkalas) [reply playbooks/roles/ocp-descheduler/tasks/main.yaml]: Used the existing role to check the cluster health status

File: playbooks/roles/ocp-descheduler/tasks/main.yaml (hunk 2 above)
Comment (AdityaHonkalas) [reply playbooks/roles/ocp-descheduler/tasks/main.yaml]: Removed this step from the tasks main playbook.

**Review by yussufsh — APPROVED** [review]
/lgtm

**Review by pravin-dsilva — APPROVED** [review]

**Pull request merged**